<?xml version="1.0"?>
<launch>

  <arg name="id_robot" default="rb1_base"/>
  <arg name="x_init_pose" default="0"/>
  <arg name="y_init_pose" default="0"/>
  <arg name="z_init_pose" default="0"/>
  <arg name="a_init_pose" default="-1.7"/>
  <arg name="load_map" default="deli7edik.yaml"/>
  <arg name="prohibition_area" default="empty_prohibition_area.yaml"/>

  <group ns="$(arg id_robot)" >

      <arg name="prefix" default="$(arg id_robot)_"/>

      <node name="map_server" pkg="map_server" type="map_server" args="$(arg load_map)">
        <param name="frame_id" value="$(arg prefix)map"/>
      </node>

      <!-- Run move_base -->
      <node pkg="move_base" type="move_base" respawn="false" name="move_base" output="screen">
        <rosparam file="$(find navigation_modes)/config/navigation/move_base_params.yaml" command="load" />
        <rosparam file="$(find navigation_modes)/config/navigation/teb_local_planner_params.yaml" command="load" />
        <rosparam file="$(find navigation_modes)/config/navigation/costmap_common_params_custom.yaml" command="load" ns="global_costmap" />
        <rosparam file="$(find navigation_modes)/config/navigation/costmap_common_params_custom.yaml" command="load" ns="local_costmap" />
        <rosparam file="$(find navigation_modes)/config/navigation/local_costmap_params.yaml" command="load" />
        <rosparam file="$(find navigation_modes)/config/navigation/global_costmap_params_map.yaml" command="load" />

        <!-- reset frame_id parameters using user input data -->
        <param name="global_costmap/obstacle_layer/base_front_laser/sensor_frame" value="$(arg prefix)front_laser_link"/>
        <param name="local_costmap/obstacle_layer/base_front_laser/sensor_frame" value="$(arg prefix)front_laser_link"/>
        <param name="local_costmap/obstacle_layer/base_front_camera/sensor_frame" value="$(arg prefix)front_rgbd_camera_depth_optical_frame"/>
        <param name="local_costmap/global_frame" value="$(arg prefix)odom"/>
        <param name="local_costmap/robot_base_frame" value="$(arg prefix)base_footprint"/>
        <param name="global_costmap/global_frame" value="$(arg prefix)map"/>
        <param name="global_costmap/robot_base_frame" value="$(arg prefix)base_footprint"/>
        <param name="TebLocalPlannerROS/map_frame" value="$(arg prefix)map"/>
        <param name="base_local_planner" value="teb_local_planner/TebLocalPlannerROS" />
        <param name="controller_frequency" value="5.0" />
        <param name="controller_patience" value="15.0" />
        <remap from="cmd_vel" to="~cmd_vel" />
        <remap from="odom" to="robotnik_base_control/odom" />
      </node>

  </group>
      
  <node name="static_tf0" pkg="tf" type="static_transform_publisher" args="0 0 0 0 0 0 /map /base_link 100"/>
    
  <!--- Run AMCL -->
  <include file="$(find navigation_modes)/launch/navigation/amcl_rb1_base.launch" >
      <arg name="id_robot" value="$(arg id_robot)"/>
      <arg name="x_init_pose" value="$(arg x_init_pose)"/>
      <arg name="y_init_pose" value="$(arg y_init_pose)"/>
      <arg name="z_init_pose" value="$(arg z_init_pose)"/>
      <arg name="a_init_pose" value="$(arg a_init_pose)"/>
  </include>

</launch>
